### PR TITLE
Update RuboCop configuration to inherit from theforeman-rubocop

### PR DESCRIPTION
**Pull request by archanaserver**
- Added `theforeman-rubocop` as a development dependency in the gemspec.
- Changed the RuboCop configuration to inherit from `default.yml` provided by `theforeman-rubocop` for better style alignment.
- Removed redundant cops as they are already included in the inherited `defsault.yml` or not needed anymore.

default.yml: https://github.com/theforeman/theforeman-rubocop/?tab=readme-ov-file#basic-style---default-performance-and-rails-cops

This is part of Rubocop standerdization, link for the reference: https://community.theforeman.org/t/standardizing-rubocop-with-theforeman-rubocop/37239

### CODE REVIEW

File: .rubocop_todo.yml [hunk 1]
```diff
@@ -276,109 +263,72 @@ Naming/FileName:
...
-  Exclude:
-    - 'test/functionals/documentation_links_test.rb'
-
-# Offense count: 2
+# Offense count: 1
 # Configuration parameters: ForbiddenMethods, AllowedMethods.
 # ForbiddenMethods: decrement!, decrement_counter, increment!, increment_counter, insert, insert!, insert_all, insert_all!, toggle!, touch, touch_all, update_all, update_attribute, update_column, update_columns, update_counters, upsert, upsert_all
 Rails/SkipsModelValidations:
   Exclude:
-    - 'db/migrate/20180605150932_deprecate_satellite_templates.rb'
```
Comment (evgeni) [inline .rubocop_todo.yml:341]: This seems to make the cop yell again :/

Comment (archanaserver) [reply .rubocop_todo.yml:341]: Probably not anymore

File: db/migrate/20180605150932_deprecate_satellite_templates.rb [hunk 2]
```diff
@@ -30,7 +30,7 @@ def remove_old_audits(destination)
   end
 
   def copy_os_default_associations(source, destination)
-    source.os_default_templates.update_all(provisioning_template_id: destination.id)
+    source.os_default_templates.update(provisioning_template_id: destination.id)
```
Comment (evgeni) [inline db/migrate/20180605150932_deprecate_satellite_templates.rb]: I do not think these two are equivalent. can't we keep that cop disabled for this file instead?

Comment (archanaserver) [reply db/migrate/20180605150932_deprecate_satellite_templates.rb]: we can, so i disabled it in recent change.

**Comment by archanaserver** [comment]
@evgeni workflow approval 👀

**Review by evgeni — APPROVED** [review]

**Pull request merged**